libvast/include/vast/detail/posix.hpp

```diff
@@ -38,7 +38,11 @@ struct uds_datagram_sender {
   static caf::expected<uds_datagram_sender> make(const std::string& path);
 
   /// Send the content of `data to `dst`.
-  caf::error send(std::span<char> data);
+  /// @param data The data that should be written.
+  /// @param timeout_usec A duration to wait for readiness of the receiver.
+  /// @returns `true` if the data was sent or `false` if it was dropped because
+  ///          the timeout was reached
+  caf::expected<bool> send(std::span<char> data, int timeout_usec = 0);
```

Comment:
I'm still unsure that `expected<bool>` is the right thing here. It feels like an anti-pattern, for two reasons:

1. There's also `caf::no_error`, which aliases to an error code with 0. The semantics are basically "not an error".
2. There is probably an error code for a timeout that makes sense to return.

Keeping it at `caf::error` and returning a timeout error code feels more appropriate. At least from the comment of the return suggests that.

Reply:
I don't have strong opinions on this though. Feel free to resolve if you think it aint't worth it to go through another round.

Reply:
To me the it makes most sense to return a 3-state type. Because I want to handle timeouts separately from errors at the call site. No strong opinion either though.

Reply:
It's a four-state type right now, though:

- true
- false
- caf::no_error
- caf::error

I think for legibility of the code it makes sense to just have this return an enum class or a variant.